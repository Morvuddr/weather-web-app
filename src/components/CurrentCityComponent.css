.currentCity{
    display: flex;
    flex-direction: column;
    width: 45%;
}

.currentCity h1 {
    text-align: left;
}

.additionalInfo {
    display: flex;
    flex-direction: row;
    align-items: center;
    align-content: space-between;
    justify-content: space-between;
}
.additionalInfo h1{
    font-size: 60px;
    margin: 0;
}
.weatherImage{
    width: 90px;
    height: 90px;
}