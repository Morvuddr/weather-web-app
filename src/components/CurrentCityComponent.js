import React from 'react';
import * as PropTypes from 'prop-types';
import '../styles/MainContainer.css';

const CurrentCityComponent = ({ weather }) => (
    !!weather ?
        <div className="currentCity">
            <h1>
                {weather.name}
            </h1>
            <div className="additionalInfo">
                <img
                    className="weatherImage"
                    src={weather.img}
                    alt='Иконка погоды'>
                </img>
                <h1>
                    {weather.temperature}
                </h1>
            </div>
        </div>
        :
        null
);

CurrentCityComponent.propTypes = {
    weather: PropTypes.object
};


export default CurrentCityComponent;