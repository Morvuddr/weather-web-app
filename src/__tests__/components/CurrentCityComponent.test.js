import CurrentCityComponent from '../../components/CurrentCityComponent';
import renderer from 'react-test-renderer';
import React from 'react';

describe('CurrentCityComponent component', () => {
    test('matches the snapshot when the weather object is not null', () => {
        const weather = {
            name: 'Moscow',
            img: 'abcd',
            temperature: '1'
        };
        const tree = renderer.create(
            <CurrentCityComponent weather={weather}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when the weather object is null', () => {
        const weather = null;
        const tree = renderer.create(
            <CurrentCityComponent weather={weather}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});