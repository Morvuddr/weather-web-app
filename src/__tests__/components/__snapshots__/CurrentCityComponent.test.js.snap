// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCityComponent component matches the snapshot when the weather object is not null 1`] = `
<div
  className="currentCity"
>
  <h1>
    Moscow
  </h1>
  <div
    className="additionalInfo"
  >
    <img
      alt="Иконка погоды"
      className="weatherImage"
      src="abcd"
    />
    <h1>
      1
    </h1>
  </div>
</div>
`;

exports[`CurrentCityComponent component matches the snapshot when the weather object is null 1`] = `null`;
